src/components/ScatterHoverLayer.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {useRef, useLayoutEffect, FunctionComponent} from 'react'
+
+import {PlotEnv} from '../utils/PlotEnv'
+import {ScatterLayerConfig} from '../types'
+import {useHoverPointIndices} from '../utils/useHoverPointIndices'
+import {drawPoints} from '../utils/drawPoints'
+import {Tooltip} from './Tooltip'
+import {getPointsTooltipData} from '../utils/tooltip'
+import {SCATTER_HOVER_POINT_SIZE} from '../constants'
+import {FILL, SYMBOL} from '../constants/columnKeys'
+
+interface Props {
+  env: PlotEnv
+  layerIndex: number
+  hoverX: number
+  hoverY: number
+}
+
+export const ScatterHoverLayer: FunctionComponent<Props> = ({
+  env,
+  layerIndex,
+  hoverX,
+  hoverY,
+}) => {
+  const layerConfig = env.config.layers[layerIndex] as ScatterLayerConfig
+  const {
+    x: xColKey,
+    y: yColKey,
+    fill: fillColKeys,
+    symbol: symbolColKeys,
+  } = layerConfig
+
+  const {xScale, yScale, innerWidth: width, innerHeight: height} = env
+  const fillScale = env.getScale(layerIndex, 'fill')
+  const symbolScale = env.getScale(layerIndex, 'symbol')
+
+  const table = env.getTable(layerIndex)
+  const xColData = table.getColumn(xColKey, 'number')
+  const yColData = table.getColumn(yColKey, 'number')
+  const fillColData = table.getColumn(FILL, 'string')
+  const symbolColData = table.getColumn(SYMBOL, 'string')
+
+  const rowIndices = useHoverPointIndices(
+    'xy',
+    hoverX,
+    hoverY,
+    xColData,
+    yColData,
+    [],
+    xScale,
+    yScale,
+    width,
+    height
+  )
+
+  const canvasRef = useRef<HTMLCanvasElement>(null)
+
+  useLayoutEffect(() => {
+    if (!canvasRef.current) {
```

Comment:
Could there be a benefit to returning if rowIndices is undefined as well? I can't quite reason about it, the functionComponent will return null, but will it still do the calculations in drawPoints?

Reply:
Agh definitely confusing. I think it's OK, since the component will return `null` so `canvasRef.current` will be falsey.